domain/consensus/model/interface_processes_headertipsmanager.go

```diff
@@ -5,5 +5,5 @@ import "github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
 // HeaderTipsManager manages the state of the header tips
 type HeaderTipsManager interface {
 	AddHeaderTip(hash *externalapi.DomainHash) error
-	SelectedTip() (*externalapi.DomainHash, error)
+	SelectedTip() (*externalapi.DomainHash, bool, error)
```

Comment:
Instead, have a function called HasSelectedTip like you've done in all the other stores.